lib/Common/Memory/PageAllocator.cpp

```diff
@@ -929,7 +929,25 @@ PageAllocatorBase<T>::FillFreePages(__in void * address, uint pageCount)
 #endif
     if (ZeroPages())
     {
-        memset(address, 0, AutoSystemInfo::PageSize * pageCount);
+        //
+        // Do memset via non-temporal store to avoid evicting existing processor cache.
+        // This helps low-end machines with limited cache size.
+        //
+#if defined(_M_IX86) || defined(_M_X64)
+        if (CONFIG_FLAG(ZeroMemoryWithNonTemporalStore))
+        {
+            size_t writeBytes = 0;
+            __m128i simdZero = _mm_setzero_si128();
+            for (__m128i * p = (__m128i *)address; writeBytes < pageCount * AutoSystemInfo::PageSize; p += 1, writeBytes += sizeof(__m128i))
+            {
+                _mm_stream_si128(p, simdZero);
+            }
+        }
+        else
+#endif
+        {
+            memset(address, 0, AutoSystemInfo::PageSize * pageCount);
+        }
```

Comment:
Can you add a NonTemporal_memset() routine for this?  Not sure where it should go... Api.h with js_memcpy_s?